ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -465,7 +465,7 @@ private void runReAct(
                     action = toolName;
 
                     if (tools.containsKey(action) && inputTools.contains(action)) {
-                        Map<String, String> toolParams = new HashMap<>();
+                        Map<String, String> toolParams = new HashMap<>(toolSpecMap.get(action).getParameters());
                         toolParams.put("input", actionInput);
```

Comment:
should we add a case for input will not be override?

Reply:
Yes, let me add one more case in next pr.